src/game.rs

```diff
@@ -0,0 +1,250 @@
+};
+
+/// Represents the state of the Game of Life.
+struct Game {
+    height: usize,
+    width: usize,
+    cells: Vec<bool>,
+}
+
+impl Game {
+    /// Creates a new Game of Life with random initial state.
+    ///
+    /// # Arguments
+    ///
+    /// * `height` - The height of the game grid.
+    /// * `width` - The width of the game grid.
+    ///
+    /// # Returns
+    ///
+    /// A new `Game` instance.
+    fn new(height: usize, width: usize) -> Self {
+        let cells = (0..height * width).map(|_| rand::random()).collect();
+        Self {
+            height,
+            width,
+            cells,
+        }
+    }
+
+    /// Gets the index in the cells vector for the given coordinates.
+    ///
+    /// # Arguments
+    ///
+    /// * `x` - The x coordinate.
+    /// * `y` - The y coordinate.
+    ///
+    /// # Returns
+    ///
+    /// The index in the cells vector.
+    fn get_index(&self, x: i32, y: i32) -> usize {
+        y as usize * self.width + x as usize
+    }
+
+    /// Counts the number of alive neighbors for the given cell.
+    ///
+    /// # Arguments
+    ///
+    /// * `x` - The x coordinate of the cell.
+    /// * `y` - The y coordinate of the cell.
+    ///
+    /// # Returns
+    ///
+    /// The number of alive neighbors.
+    fn count_neighbors(&self, x: i32, y: i32) -> usize {
+        let mut count = 0;
+        for dx in -1..=1 {
+            let nx = x + dx;
+            if nx < 0 || nx >= self.width as i32 {
+                // checks if neighbor's x is out of bounds
+                continue;
+            }
+            for dy in -1..=1 {
+                let ny = y + dy;
+                if ny < 0 || ny >= self.height as i32 {
+                    // checks if neighbor's y is out of bounds
+                    continue;
+                }
+                if nx == x && ny == y {
+                    continue;
+                }
+                let index = self.get_index(nx, ny);
+                if self.cells[index] {
+                    count += 1;
+                }
+            }
+        }
+        count
+    }
+
+    /// Updates the game state to the next generation.
+    fn update(&mut self) {
+        let mut next_cells = vec![false; self.height * self.width];
+        for x in 0..self.width as i32 {
+            for y in 0..self.height as i32 {
+                let index = self.get_index(x, y);
+                let cell = self.cells[index];
+                let neighbors = self.count_neighbors(x, y);
+
+                next_cells[index] = matches!((cell, neighbors), (true, 2) | (true, 3) | (false, 3));
+            }
+        }
+        self.cells = next_cells
+    }
+
+    /// Draws the current game state.
+    ///
+    /// # Arguments
+    ///
+    /// * `cell_size` - The size of each cell.
+    fn draw(&self, cell_size: f32) {
+        for x in 0..self.width {
+            for y in 0..self.height {
+                let index = y * self.width + x;
```

Comment:
shouldn't you use the `get_index` here?

Reply:
Yes, I changed `get_index`'s original code and forgot to update this part. Anyways, I'll change `cells` to be a Vec<Vec<bool> so that indexing function won't be needed anymore.